WcaOnRails/app/models/competition.rb

```diff
@@ -458,11 +465,82 @@ def in_progress?
   end
 
   def is_over?
-    end_date < Date.today
+    !!end_date && end_date < Date.today
+  end
+
+  def results_updated_at
+    Option.from_nilable(results.order('updated_at desc').pluck(:updated_at).first)
+  end
+
+  def result_cache_key(view)
+    [id, view, results_updated_at.map(&:iso8601).unwrap_or("")]
```

Comment:
I'm having trouble understanding the motivation of this Option class. This feels like an idiom that you are familiar with that I've never seen before. Is the idea to deal with Competitions that don't have results without having to write an explicit if statement for them?